File: hw/toapanta/u3/hw33Strategy/src/main/java/ec/edu/espe/hwstrategy/model/SortingContext.java
package ec.edu.espe.hwstrategy.model;

public class SortingContext {
    private SortingStrategy ss;

    public int[] sort(int data[]) {
        int size = data.length;
        ss = setSortStrategy(size);
        return ss.sort(data);
    }

    // Select Strategy
    public SortingStrategy setSortStrategy(int n) {
        if (n > 0 && n < 30) {
            return new BubbleSort();
        } else if (n >= 30 && n < 100) {
            return new InsertionSort();
        } else {
            return new QuickSort();
        }
    }
    }

File: exam/ramos/u3/python/model/sorting_context.py
from model.bubble_sort import BubbleSort
from model.insertion_sort import InsertionSort
from model.quick_sort import QuickSort

class SortingContext:

    def __init__(self):
        self.ss = None

    def sort(self, data):
        size = len(data)
        self.ss = self.set_sort_strategy(size)
        return self.ss.sort(data)

    def set_sort_strategy(self, n):
        if 2 < n < 6:
            return BubbleSort()
        if 6 <= n < 10:
            return InsertionSort()
        if n >= 10:
            return QuickSort()

    def get_strategy_name(self):
        if isinstance(self.ss, BubbleSort):
            return "BubbleSort"
        if isinstance(self.ss, InsertionSort):
            return "InsertionSort"
        if isinstance(self.ss, QuickSort):
            return "QuickSort"
        return "Unknown"
